drivers/eeprom/include/eeprom_types.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2018 Inria
+ *               2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup    drivers_eeprom
+ *
+ * @{
+ *
+ * @file
+ * @brief       Types definitions used in the EEPROM API
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef EEPROM_TYPES_H
+#define EEPROM_TYPES_H
+
+#include "board.h"
+#include "eeprom.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(HAVE_EEPROM_T) || defined(DOXYGEN)
+/**
+ * @brief   Device handle type for EEPROM devices
+ */
+typedef uint_fast8_t eeprom_t;
+#endif
+
+#if !defined(HAVE_EEPROM_OFF_T) || defined(DOXYGEN)
+/**
+ * @brief   Offset / size type in EEPROM devices
+ */
+typedef uint32_t eeprom_off_t;
+#endif
+
+/**
+ * @brief   Driver interface for EEPROM devices
+ *
+ * This interface must be implemented by every EEPROM device driver. The
+ * function pointer in @ref eeprom_driver_t::erase can be `NULL`.
+ */
+typedef struct {
+    /**
+     * @brief   Initialize the data structures for the EEPROM device
+     *
+     * @param[in,out]   handle  Device handle to initialize
+     * @param[in]       params  Initialization parameters of the device
+     *
+     * @retval  0       Failed to initialize the device
+     * @return          The size of the device in bytes
+     */
+    eeprom_off_t (*init)(void *handle, const void *params);
```

Comment:
Has this still to be part of the driver. If I'm right, a device that is registered has to be initialized by the application or `board_init` function. That would be different, if you would define a static list of devices which are initialized by the `init` function of this API.

Reply:
No, it is a leftover I overlooked

Reply:
Removed